deepchecks/tabular/checks/model_evaluation/weak_segments_performance.py

```diff
@@ -0,0 +1,328 @@
+            classes_index_order: Union[np.array, pd.Series, None] = None,
+            n_samples: int = 10_000,
+            categorical_aggregation_threshold: float = 0.05,
+            n_to_show: int = 3,
+            random_state: int = 42,
+            **kwargs
+    ):
+        super().__init__(**kwargs)
+        self.columns = columns
+        self.ignore_columns = ignore_columns
+        self.n_top_features = n_top_features
+        self.segment_minimum_size_ratio = segment_minimum_size_ratio
+        self.n_samples = n_samples
+        self.n_to_show = n_to_show
+        self.random_state = random_state
+        self.loss_per_sample = loss_per_sample
+        self.classes_index_order = classes_index_order
+        self.user_scorer = alternative_scorer if alternative_scorer else None
+        self.categorical_aggregation_threshold = categorical_aggregation_threshold
+
+    def run_logic(self, context: Context, dataset_kind) -> CheckResult:
+        """Run check."""
+        dataset = context.get_data_by_kind(dataset_kind)
+        dataset.assert_features()
+        dataset = dataset.sample(self.n_samples, random_state=self.random_state, drop_na_label=True)
+        predictions = context.model.predict(dataset.features_columns)
+        y_proba = context.model.predict_proba(dataset.features_columns) if \
+            context.task_type in [TaskType.MULTICLASS, TaskType.BINARY] else None
+
+        if self.loss_per_sample is not None:
+            loss_per_sample = self.loss_per_sample[list(dataset.data.index)]
+        else:
+            loss_per_sample = calculate_per_sample_loss(context.model, context.task_type, dataset,
+                                                        self.classes_index_order)
+        dataset = dataset.select(self.columns, self.ignore_columns, keep_label=True)
+        if len(dataset.features) < 2:
+            raise DeepchecksNotSupportedError('Check requires data to have at least two features in order to run.')
+        encoded_dataset = self._target_encode_categorical_features_fill_na(dataset)
+        dummy_model = _DummyModel(test=encoded_dataset, y_pred_test=predictions, y_proba_test=y_proba,
+                                  validate_data_on_predict=False)
+
+        if context.features_importance is not None:
+            feature_rank = context.features_importance.sort_values(ascending=False).keys()
+            feature_rank = np.asarray([col for col in feature_rank if col in encoded_dataset.features])
+        else:
+            feature_rank = np.asarray(encoded_dataset.features)
+
+        scorer = context.get_single_scorer(self.user_scorer)
+        weak_segments = self._weak_segments_search(dummy_model, encoded_dataset, feature_rank,
+                                                   loss_per_sample, scorer)
+        if len(weak_segments) == 0:
+            raise DeepchecksProcessError('Unable to train a error model to find weak segments.')
```

Comment:
```suggestion
            raise DeepchecksProcessError('WeakSegmentsPerformance was unable to train an error model to find weak segments. This might mean that ... .. ...')
```

The user should understand what this means. Could this mean there are no weak segments? The data is bad? The user is bad and is going to hell?